### DISCUSSION TITLE
I can't connect to java server with Io Client socket

### DISCUSSION THREAD

**Discussion by ebrahemimorteza** [socketio/socket.io#4912] [Q&A]
I use url http://localhost:8080/LAS/chat/*  but I get every time 404, it doesn't seem to recognize my project name or my address . This is my problem    WebSocket connection to 'ws://localhost:8080/socket.io/?chatId=&username=90310995470183035401&idPackage=23&idLesson=27&idReciveer=&user=1&server=&EIO=4&transport=websocket' failed: Error during WebSocket handshake: Unexpected response code: 404

**Comment by darrachequesne** [comment]
You need to make sure the request paths match on both side.

For example, if your server is served at `http://localhost:8080/LAS/chat/`, then you need to create your client with the `/LAS/chat/` path:

```js
const socket = io("http://localhost:8080", {
  path: "/LAS/chat/" // default to "/socket.io/"
});
```

Reference: https://socket.io/docs/v4/client-options/#path
